scripts/release.ts

```diff
@@ -0,0 +1,191 @@
+  console.log(chalk.green('Creating draft PR...'))
+  // TODO(0xdef1cafe): remove logging
+  console.log(chalk.yellow(command))
+  // TODO(0xdef1cafe): uncomment
+  // await pify(exec)(command)
+  console.log(chalk.green('Draft PR created.'))
+  exit(chalk.green(`Release ${nextVersion} created.`))
+}
+
+type GetCommitMessagesArgs = 'develop' | 'release'
+type GetCommitMessagesReturn = {
+  messages: string[]
+  total: number
+}
+type GetCommitMessages = (branch: GetCommitMessagesArgs) => Promise<GetCommitMessagesReturn>
+const getCommits: GetCommitMessages = async branch => {
+  const { all, total } = await git().log([
+    '--oneline',
+    '--first-parent',
+    '--pretty=format:%s', // no hash, just conventional commit style
+    `origin/main..origin/${branch}`,
+  ])
+
+  const messages = all.map(({ hash }) => hash)
+  return { messages, total }
+}
+
+const assertCommitsToRelease = (total: number) => {
+  if (!total) exit(chalk.red('No commits to release.'))
+}
+
+const doRegularRelease = async () => {
+  await fetch()
+  const { messages, total } = await getCommits('develop')
+  assertCommitsToRelease(total)
+  await inquireProceedWithCommits(messages)
+  console.log(chalk.green('Checking out develop...'))
+  await git().checkout(['develop'])
+  console.log(chalk.green('Pulling develop...'))
+  await git().pull()
+  console.log(chalk.green('Resetting release to develop...'))
+  await git().checkout(['-B', 'release']) // reset release to develop
+  console.log(chalk.green('Force pushing release branch...'))
+  // TODO(0xdef1cafe): remove --dry-run
+  await git().push(['--dry-run', '--force', 'origin', 'release'])
+  await createDraftPR()
+  exit()
+}
+
+const doHotfixRelease = async () => {
+  exit(chalk.yellow('Unimplemented. PRs welcome!'))
+  // TODO(0xdef1cafe): implement hotfix release
+  // 1. ask if we want to merge currently checked out branch to main
+  // 2. set release to current branch
+  // 3. force push release to origin
+  // 4. create draft PR
+}
+
+const getSemverTags = async (): Promise<string[]> => {
+  const tags = await pify(gitSemverTags)()
+  if (!tags.length) exit(chalk.red('No semver release tags found.'))
+  return tags
+}
+
+const getLatestSemverTag = async (): Promise<string> => {
+  const tags = await getSemverTags()
+  const tag = tags[0]
+  semver.valid(tag) || exit(chalk.red(`${tag} is not a valid semver tag.`))
+  return tags[0]
+}
+
+type WebReleaseType = Extract<semver.ReleaseType, 'minor' | 'patch'>
+
+const getNextReleaseVersion = async (versionBump: WebReleaseType): Promise<string> => {
+  const latestTag = await getLatestSemverTag()
+  const nextVersion = semver.inc(latestTag, versionBump)
+  if (!nextVersion) exit(chalk.red(`Could not bump version to ${nextVersion}`))
+  return nextVersion!
+}
+
+const assertGhInstalled = async () => {
+  try {
+    await pify(exec)('hash gh') // will throw if gh is not installed
```

Comment:
change this to something like `hash foo` to test

Reply:
Confirmed working with `'hash dajlnffdaf'`